packages/server/src/api/controllers/query/index.ts

```diff
@@ -142,6 +158,66 @@ export async function preview(ctx: UserCtx) {
 
   const authConfigCtx: any = getAuthConfig(ctx)
 
+  function getFieldMetadata(field: any, key: string): QuerySchema {
+    const makeQuerySchema = (
+      type: FieldType,
+      name: string,
+      subtype?: string
+    ): QuerySchema => ({
+      type,
+      name,
+      subtype,
+    })
+    let type = typeof field,
+      fieldMetadata = makeQuerySchema(FieldType.STRING, key)
+    if (field != null)
```

Comment:
So if I'm following the logic correctly, we're allowing `field` to be `null` or `undefined` and in these cases, we assume it's a string. Is that correct?

If so, this feels like a bug waiting to happen. Would it not be better to raise an error when `field` is `null` or `undefined`, and do a strict check for the `case "string"`, in the spirit of blowing up as early as possible when something we don't expect to happen happens?

Reply:
That's right, if a field has `null` or `undefined` in the query response we assume it is string field type.

I believe we have to do this because we're trying to build a schema from dynamic input. If the value is null/undefined, assuming that the type is a string is the safest choice. There is no Unknown field type. 

I don't think we'd ever want to throw an error because we generate the schema from the first row:

```ts
const fieldMetadata = getFieldMetadata(rows[0][key], key)
```

Reply:
That makes sense. Could you leave a comment explaining that this is what we're doing next to the string assumption? 🙏